### PR TITLE
[CELEBORN-1573] Change to debug logging on client side for reserve slots

### PR DESCRIPTION
**Pull request by akpatnam25**
### What changes were proposed in this pull request?
Change client side reserve slots log to debug. 

### Why are the changes needed?
in a large fleet of workers such as ours, it is noisy for the user in the driver logs. 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
N/A

### CODE REVIEW

**Comment by akpatnam25** [comment]
cc @FMX @SteNicholas @mridulm @zhouyejoe

**Review by zhouyejoe — APPROVED** [review]
LGTM

File: client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala [hunk 1]
```diff
@@ -651,7 +651,7 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
         replyRegisterShuffle(RegisterShuffleResponse(StatusCode.SLOT_NOT_AVAILABLE, Array.empty))
         return
       case StatusCode.SUCCESS =>
-        logInfo(s"OfferSlots for $shuffleId Success!Slots Info: ${res.workerResource}")
+        logDebug(s"OfferSlots for $shuffleId Success!Slots Info: ${res.workerResource}")
```
Comment (cxzl25) [inline client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala:654]:
```suggestion
        if (log.isDebugEnabled()) {
          logDebug(s"OfferSlots for $shuffleId Success!Slots Info: ${res.workerResource}")
        } else {
          logInfo(
            s"OfferSlots for $shuffleId Success!Slots Info size: ${res.workerResource.size()}")
        }
```

**Review by cxzl25 — APPROVED** [review]

File: client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala (hunk 1 above)
Comment (FMX) [reply client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala:654]: I disagree with this suggestion.

**Comment by s0nskar** [comment]
I think this change can make retrospective debugging hard as we will no insight into which workers the shuffle data was written to, what was the behaviour of workers during that time. 

Instead of making it completely a debug log IMO we should at least output some basic info about the worker host selected for shuffle this will reduce the noise but also leave some space for debugging issues.

Ex –
```
Host: 10.152.63.41
WorkerStatus: WorkerStatus{state=Normal, stateStartTime=1724244049248}
NumOfSlotsAllocated: 100
```

**Comment by akpatnam25** [comment]
> I think this change can make retrospective debugging hard as we will have no insight into which workers the shuffle data was written to and what was the behaviour of those workers during that time.
> 
> Instead of making it completely a debug log IMO we should at least output some basic info about the worker host selected for shuffle this will reduce the noise but also leave some space for debugging issues.
> 
> Ex –
> 
> ```
> Host: 10.152.63.41
> WorkerStatus: WorkerStatus{state=Normal, stateStartTime=1724244049248}
> NumOfSlotsAllocated: 100
> ```

I think this will still be noisy in a large fleet, especially in the driver logs where users might be looking for other logs/errors. I feel that if truly needed for debugging, the user can then enable debug logging for it.

**Comment by FMX** [comment]
> > I think this change can make retrospective debugging hard as we will have no insight into which workers the shuffle data was written to and what was the behaviour of those workers during that time.
> > Instead of making it completely a debug log IMO we should at least output some basic info about the worker host selected for shuffle this will reduce the noise but also leave some space for debugging issues.
> > Ex –
> > ```
> > Host: 10.152.63.41
> > WorkerStatus: WorkerStatus{state=Normal, stateStartTime=1724244049248}
> > NumOfSlotsAllocated: 100
> > ```
> 
> I think this will still be noisy in a large fleet, especially in the driver logs where users might be looking for other logs/errors. I feel that if truly needed for debugging, the user can then enable debug logging for it.

Maybe print this log to one line can be helpful.

**Comment by FMX** [comment]
> > > I think this change can make retrospective debugging hard as we will have no insight into which workers the shuffle data was written to and what was the behaviour of those workers during that time.
> > > Instead of making it completely a debug log IMO we should at least output some basic info about the worker host selected for shuffle this will reduce the noise but also leave some space for debugging issues.
> > > Ex –
> > > ```
> > > Host: 10.152.63.41
> > > WorkerStatus: WorkerStatus{state=Normal, stateStartTime=1724244049248}
> > > NumOfSlotsAllocated: 100
> > > ```
> > 
> > 
> > I think this will still be noisy in a large fleet, especially in the driver logs where users might be looking for other logs/errors. I feel that if truly needed for debugging, the user can then enable debug logging for it.
> 
> Maybe print this log to one line can be helpful.

Setting the logger levels for components could be the ultimate solution.

**Review by FMX — APPROVED** [review]
LGTM. Merged into main(v0.6.0), branch-0.5(v0.5.2), branch-0.4(v0.4.3).

**Comment by s0nskar** [comment]
> Maybe print this log to one line can be helpful.

@FMX i created a PR sometime back to support single line logging – https://github.com/apache/celeborn/pull/2672. Let me wdyt about this, i'll extend it to other places.

**Pull request closed without merging**